app/controllers/users/registrations_controller.rb

```diff
@@ -30,9 +30,23 @@ class Users::RegistrationsController < Devise::RegistrationsController
   # end
 
   # DELETE /resource
-  # def destroy
-  #   super
-  # end
+  def destroy
+    # Ensure user is authenticated
+    unless current_user
+      redirect_to new_user_session_path, alert: "You must be logged in to delete your account."
```

Comment:
Can you make these strings internationalized?  This simply means that the strings have to be wrapped with `_('source string')`.